### ISSUE TITLE
Reimplement Transaction Fee Distribution.

### ISSUE THREAD

**Issue by 0o-de-lally** [OLSF/libra#53]
TLDR;
Libra has two conflicting models of transaction fee distribution. In the stdlib folder it seems the fees are burnt, 

In the prover test suite, it is instead transferred to validators. This is the model OL needs to implement.

As implemented in /move-prover/ tests (how OL should do this).

https://github.com/OLSF/libra/blob/02acecbacb3ba40e55d3dd20f26ce78d8b4e8c86/language/move-prover/tests/sources/stdlib/modules/transaction_fee.move

-- This code is likely correct, but we need the distribution of fees to be set to Node Weight, and not equal.

How it appears in /stdlib/:
https://github.com/OLSF/libra/blob/02acecbacb3ba40e55d3dd20f26ce78d8b4e8c86/language/stdlib/modules/TransactionFee.move
